sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs

```diff
@@ -29,9 +29,640 @@ public class UnusedPrivateMemberTest
     {
         [TestMethod]
         [TestCategory("Rule")]
-        public void UnusedPrivateMember()
+        public void Field_Accessibility()
         {
-            Verifier.VerifyAnalyzer(@"TestCases\UnusedPrivateMember.cs", new UnusedPrivateMember());
+            Verifier.VerifyCSharpAnalyzer(@"
+public class PrivateMembers
+{
+    private int privateField; // Noncompliant {{Remove the unused private field 'privateField'.}}
+//  ^^^^^^^^^^^^^^^^^^^^^^^^^
+    private static int privateStaticField; // Noncompliant
+
+    private class InnerPrivateClass // Noncompliant
+    {
+        internal int internalField; // Noncompliant
+        protected int protectedField; // Noncompliant
```

Comment:
Do we report on unused _protected_ methods in internal classes?

Reply:
No, we report on members with "effective" visibility "private" or "internal" when they are classes.

Reply:
"Effective" is computed visibility